db/changes.go

```diff
@@ -494,20 +532,33 @@ func (db *DatabaseCollectionWithUser) changesFeed(ctx context.Context, singleCha
 	return feed
 }
 
-func makeChangeEntry(logEntry *LogEntry, seqID SequenceID, channel channels.ID) ChangeEntry {
+func makeChangeEntry(ctx context.Context, logEntry *LogEntry, seqID SequenceID, channel channels.ID, versionType ChangesVersionType) ChangeEntry {
 	change := ChangeEntry{
 		Seq:          seqID,
 		ID:           logEntry.DocID,
 		Deleted:      (logEntry.Flags & channels.Deleted) != 0,
-		Changes:      []ChangeRev{{"rev": logEntry.RevID}},
+		Changes:      []ChangeByVersionType{{ChangesVersionTypeRevTreeID: logEntry.RevID}},
 		branched:     (logEntry.Flags & channels.Branched) != 0,
 		principalDoc: logEntry.IsPrincipal,
 		collectionID: logEntry.CollectionID,
 	}
+
+	switch versionType {
+	case ChangesVersionTypeCV:
+		if logEntry.SourceID != "" {
+			change.Changes[0] = ChangeByVersionType{versionType: Version{SourceID: logEntry.SourceID, Value: logEntry.Version}.String()}
+		}
+	case ChangesVersionTypeRevTreeID:
+		fallthrough
+	default:
+		// already initialized with a 'rev' change entry
+	}
+
 	// populate CurrentVersion entry if log entry has sourceID and Version populated
 	// This allows current version to be nil in event of CV not being populated on log entry
 	// allowing omitempty to work as expected
 	if logEntry.SourceID != "" {
+		// TODO: Remove this if we change BLIP generateBlipSyncChanges and tests to use versionType and read the value from the normal Changes array... no reason to store this info in two places.
```

Comment:
I think this TODO should get a ticket if we want to do this - and it could be marked in `type Change`.